packages/test/src/test-integration-workflows.ts

```diff
@@ -1274,32 +1275,31 @@ test('Count workflow executions', async (t) => {
   const worker = await createWorker();
   const client = t.context.env.client;
 
-  // Run 2 workflows that don't complete
-  // (use startWorkflow to avoid waiting for workflows to complete, which they never will)
-  for (let i = 0; i < 2; i++) {
-    await startWorkflow(completableWorkflow, { args: [false] });
-  }
-
   await worker.runUntil(async () => {
-    // Run 3 workflows that complete.
     await Promise.all([
+      // Run 2 workflows that will never complete...
+      startWorkflow(completableWorkflow, { args: [false] }),
+      startWorkflow(completableWorkflow, { args: [false] }),
+
+      // ... and 3 workflows that will complete
       executeWorkflow(completableWorkflow, { args: [true] }),
       executeWorkflow(completableWorkflow, { args: [true] }),
       executeWorkflow(completableWorkflow, { args: [true] }),
     ]);
   });
 
+  // FIXME: Find a better way to wait for visibility to stabilize
+  await setTimeoutPromise(1000);
```

Comment:
I thought there was some kind of `assertEventually` ?

Reply:
We do, but it is awkward because we can't use `t.deepEquals()` from it. Sleeping for 1s isn't that bad anyway, given that those tests execute concurrently anyway, so that's not blocking.

Reply:
But we have that need of "waiting for vis to catch up" in a few places, and so I would expect users. So I want to take time at some point to see if we could somehow add an API for that as part of the Dev/Test server environment API to wait for visibility to sync.